### DISCUSSION TITLE
Meanings behind page.annot_xrefs() and document.xref_length()

### DISCUSSION THREAD

**Discussion by chaugong** [pymupdf/PyMuPDF#3271] [Looking for help]
I don't really understand the two functions stated in title.

For example, i have a pdf (only one page). And Document.xref_length() gives me 31. It means there are 31 COS objects, right?
However, when i run page.annot_xrefs(), it only gives me a list of 2 tuples (I expect to be 31 tuples). 

SO i want to ask what are the differences in the meaning of the two number (31 & 2)?

**Comment by JorjMcKie** [comment]
`Document.xref_length()` gives you the length of the table of cross reference numbers in the **_complete document_**. Each of these numbers represents a pointer to one of the document's objects.

The list `page.annot_xrefs()` contains object numbers of the **_annotations_** of that page. Obviously, your page has 2 annotation items.
But the whole document has 29 more / other objects.

Does this help?
